open-api/shared-ide/shared-ide.gradle

```diff
@@ -1,3 +1,7 @@
+plugins {
+    id 'io.deephaven.project.java-public'
```

Comment:
afaict, this stuff needs to be public. @niloc132 for further clarification. I think some of this stuff could be cleaned up, but out of scope for this PR IMO.

Reply:
Some of these are shared, and depended on (eventually) by the server so it can do autocomplete work. In DHE we actually shared that code with the client in a way that would pass objects between client and server, but in DHC we use gRPC instead. https://github.com/deephaven/deephaven-core/issues/187 is tracking the removal of those actual shared types, we probably need to merge some of the empty/pointless projects too.